reana_commons/publisher.py

```diff
@@ -146,31 +146,3 @@ def publish_workflow_status(self, workflow_uuid, status, logs="", message=None):
             "message": message,
         }
         self._publish(msg)
-
-
-class WorkflowSubmissionPublisher(BasePublisher):
```

Comment:
I agree that it's only used in `reana-server`, but in my view it was not necessary to move workflow submission publisher elsewhere. In this file we have a base class and `WorkflowStatusPublisher` as another implementation for reference. Also, moving into `reana-server` under `scheduler` may not be the best place because `scheduler.py` (now `scheduler/consumer.py`) used to be a file which lives in a separate container under `reana-server` deployment. Now it is mixed which might be a bit more confusing.
@tiborsimko , @mvidalgarcia , it would be interesting to hear your opinions

Reply:
Regarding removing publisher, in this case, we will need to release a new version of `reana-commons` as I added `retry_count` argument to a publisher. So a small change in the scheduler in one repo triggers the release of a library in the other one. This feels painful. Code that changes together, should stay together.

Reply:
> moving into reana-server under scheduler may not be the best place because scheduler.py (now scheduler/consumer.py) used to be a file which lives in a separate container under reana-server deployment

Took a bit to understand what you meant. I see your point. Yeah, I can move it to `publisher.py` in `reana_server/`

Reply:
I moved `scheduler.py` back and created a `publisher.py` file for WorkflowSubmissionPublisher.